### PR TITLE
Create Cloud Build CD job

### PR DESCRIPTION
**Pull request by kkr16**
Moved CD jobs for staging to Cloud Build.

- Bumped tf providers' versions for Google Cloud to be able to fix bugs related to Cloud Build resources on older tf provider
- When a PR is opened:
  - The `pr-open.yaml` job will run build and stage Gcping to `https://pr-${PR_NUMBER}-us-central1-5tkroniexa-uc.a.run.app/` (e.g. for this PR https://pr-135-us-central1-5tkroniexa-uc.a.run.app/)
  - `terraform plan` to provide a preview of the production changes this PR will effect when it is applied
- When a commit is pushed to main (via PR or other):
  - the `deploy.yaml` job will run, will build the image and tag it with `latest`, then will deploy it to all endpoints in production
- Minor changes to manifests to accommodate bumped tf versions (e.g. setting `location = "US"` since it is no longer a default with newer version of the tf provides for GCP
- Minor changes to manifests to make deploying to other environments easier

### CODE REVIEW

File: tools/cloudbuild/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  - ko-build
+
+- name: 'gcr.io/${PROJECT_ID}/terraform'
+  id: terraform-plan-preview
+  entrypoint: /bin/bash
+  args:
+    - -c
+    - |
+      terraform -chdir=tools/terraform apply \
+      -auto-approve \
```
Comment (enocom) [inline tools/cloudbuild/deploy.yaml:56]: Should this be `--auto-approve`?

File: tools/cloudbuild/pr-open.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    - -c
+    - |
+      gcloud run services add-iam-policy-binding pr-${_PR_NUMBER}-us-central1 \
+      --member="allUsers" \
+      --role="roles/run.invoker" \
+      --region=us-central1
+ 
+- name: 'gcr.io/${PROJECT_ID}/terraform'
+  id: terraform-init
+  args: ['-chdir=tools/terraform', 'init', '-reconfigure']
```
Comment (enocom) [inline tools/cloudbuild/pr-open.yaml:66]: nit: `--chdir` and `--reconfigure`?

File: tools/terraform/gcb_iam.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  service = "iam.googleapis.com"
+}
+
+// Create custom Service Account for Cloud Build
+resource "google_service_account" "gcb" {
+  account_id   = "cloudbuild"
+  display_name = "Service account for Cloud Build jobs"
+}
+
+resource "google_project_iam_member" "gcb-run-admin" {
```
Comment (enocom) [inline tools/terraform/gcb_iam.tf:26]: nit: gcb-run-admin should be `gcp_run_admin` to match resource name. Ditto below.

**Review by enocom — CHANGES_REQUESTED** [review]
A few minor nits. Looks great!

File: tools/cloudbuild/deploy.yaml (hunk 1 above)
Comment (kkr16) [reply tools/cloudbuild/deploy.yaml:56]: tf [documentation](https://developer.hashicorp.com/terraform/cli/commands/apply#auto-approve) mentions `-auto-approve` but `--auto-approve` is also accepted.

File: tools/cloudbuild/pr-open.yaml (hunk 2 above)
Comment (kkr16) [reply tools/cloudbuild/pr-open.yaml:66]: Same as previous comment.

File: tools/terraform/gcb_iam.tf (hunk 3 above)
Comment (kkr16) [reply tools/terraform/gcb_iam.tf]: Fixed! Thank you! :)

File: tools/cloudbuild/pr-open.yaml (hunk 2 above)
Comment (kkr16) [reply tools/cloudbuild/pr-open.yaml:66]: I ended up changing two instances of `--var` to use `-var` instead, for consistency across our project, and to make sure we follow tf docs.

**Comment by kkr16** [comment]
> A few minor nits. Looks great!

Thanks for your review @enocom ! Back to you. :)

**Review by enocom — APPROVED** [review]

**Pull request merged**